src/lms-ds-loader/edfi_lms_ds_loader/mssql_lms_operations.py

```diff
@@ -210,3 +215,15 @@ def soft_delete_from_production(self, table: str, sourceSystem: str) -> None:
 
         rowcount = self._exec(statement)
         logger.debug(f"Soft-deleted {rowcount} records in table `{table}`")
+
+    def get_processed_files(self, resource_name: str) -> pd.DataFrame:
+        query = f"""
+SELECT FullPath FROM lms.ProcessedFiles
+WHERE ResourceName = '{resource_name}'""".strip()
+        return self._query(query)
```

Comment:
Using the DataFrame on line 53 was a nice convenience to avoid writing ORM code and to avoid having to manually iterate over rows and build a list. But in the end it will be easier to deal with the list - or in fact as mentioned above, a set - of file names

```suggestion
    def get_processed_files(self, resource_name: str) -> Set[str]:
        query = f"""
SELECT FullPath FROM lms.ProcessedFiles
WHERE ResourceName = '{resource_name}'""".strip()

        df = pd.read_sql_query(query, self.engine)

       return set(df["FullPath"])
```